[1498] Replace availability calendar with FullCalendar [v5.5]

orenyk opened the pull request:
Resolves #1498 on release-v5.5
- install FullCalendar v2.3.1 (blocked on Rails 4.1)
- generate event hash in controller
- generate event array in JS for speed
- remove old JS calendar

This is a bit hack-y and probably could be better, looking forward to some good code review :stuck_out_tongue:. I'll work on a much cleaner version of this for `master` once we've merged in #1360.

esoterik commented:
the days in the past should be greyed out

esoterik commented:
Suggestion: color the box yellow when only one item is available for that day

esoterik commented on app/controllers/equipment_models_controller.rb:
```diff
@@ -184,4 +191,43 @@ def fix_content_type(filedata)
     Paperclip::FileCommandContentTypeDetector.new(file).detect
   end
+
+  def calculate_availability # rubocop:disable all
```
can we use existing model methods for this? seems pretty similar

esoterik commented on app/controllers/equipment_models_controller.rb:
```diff
@@ -184,4 +191,43 @@ def fix_content_type(filedata)
+    max_avail = @equipment_model.equipment_items.active.count
+
+    @avail_data = []
+    (@start_date..@end_date).map do |date|
```
we definitely have existing count methods

esoterik commented:
disregard my inline comments!

orenyk commented:
Dammit I forgot to add / commit the JSON builder file for this branch and just deleted it locally. @esoterik did you actually test this locally?

orenyk commented:
Ok, I cleaned up a bit and made sure that past days would be colored grey. In terms of yellow coloring when there's only one, I implemented it but I'm not sure it makes sense given that if there's only one item to begin with then it's not really "low", just normal (it colors the cell yellow if there are less than 1/4 of the items available, which isn't great but is something). See what you think and I'll consider ways to make the coloring better.

orenyk commented:
Never mind, made the colors better :smile:, you'll have to check it out on equipment with 3 or more items to see the range.

esoterik commented:
okay, this looks good!

orenyk commented:
Also reviewed by @AlanLiu96, merging!
